src/main/java/com/evuv/expressions/CollectionValueExpression.java

```diff
@@ -0,0 +1,43 @@
+package com.evuv.expressions;
+
+import com.evuv.exceptions.EventBindingException;
+
+import java.util.Collection;
+import java.util.Map;
+
+public class CollectionValueExpression <T extends Collection> implements Expression<T>, BindedExpression<T> {
+    private T value;
+
+    public CollectionValueExpression(T val) {
+        this.value = val;
+    }
+    @Override
+    public T getValue() {
+        return value;
+    }
+
+    @Override
+    public Expression getLeft() {
+        return null;
+    }
+
+    @Override
+    public Expression getRight() {
+        return null;
+    }
+
+    @Override
+    public void setLeft(Expression left) {
+        
```

Comment:
maybe we should throw here operation not supported exception ? to catch error in the parser ?

Reply:
It will be inconsistent with rest classes behavior.

Reply:
agree.